docs/source/config_file.rst

```diff
@@ -65,234 +65,346 @@ unfortunate, and is subject to change in future versions.
+    name of the *imported* module, not the module containing the import
+    statement.
 
-.. note::
+``follow_imports`` (string, default ``normal``) 
+    Directs what to do with imports when the imported module is found
+    as a ``.py`` file and not part of the files, modules and packages 
+    provided on the command line.
 
-   If multiple pattern sections match a module, the options from the
-   most specific section are used where they disagree.  This means
-   that ``foo.bar`` will take values from sections with the patterns
-   ``foo.bar``, ``foo.bar.*``, and ``foo.*``, but when they specify
-   different values, it will use values from ``foo.bar`` before
-   ``foo.bar.*`` before ``foo.*``.
+    The four possible values are ``normal``, ``silent``, ``skip`` and
+    ``error``.  For explanations see the discussion for the
+    :ref:`--follow-imports <follow-imports>` command line flag. 
+    
+    Note that if pattern matching is used, the pattern should match the
+    name of the *imported* module, not the module containing the import
+    statement.
 
-- ``follow_imports`` (string, default ``normal``) directs what to do
-  with imports when the imported module is found as a ``.py`` file and
-  not part of the files, modules and packages on the command line.
-  The four possible values are ``normal``, ``silent``, ``skip`` and
-  ``error``.  For explanations see the discussion for the
-  :ref:`--follow-imports <follow-imports>` command line flag.  Note
-  that if pattern matching is used, the pattern should match the name
-  of the *imported* module, not the module containing the import
-  statement.
+``follow_imports_for_stubs`` (boolean, default False)
+    Determines whether to respect the ``follow_imports`` setting even for
+    stub (``.pyi``) files.
 
-- ``follow_imports_for_stubs`` (Boolean, default false) determines
-  whether to respect the ``follow_imports`` setting even for stub
-  (``.pyi``) files.
-  Used in conjunction with ``follow_imports=skip``, this can be used
-  to suppress the import of a module from ``typeshed``, replacing it
-  with `Any`.
-  Used in conjunction with ``follow_imports=error``, this can be used
-  to make any use of a particular ``typeshed`` module an error.
+    Used in conjunction with ``follow_imports=skip``, this can be used
+    to suppress the import of a module from ``typeshed``, replacing it
+    with `Any`.
 
-- ``ignore_missing_imports`` (Boolean, default False) suppress error
-  messages about imports that cannot be resolved.  Note that if
-  pattern matching is used, the pattern should match the name of the
-  *imported* module, not the module containing the import statement.
+    Used in conjunction with ``follow_imports=error``, this can be used
+    to make any use of a particular ``typeshed`` module an error.
 
-- ``silent_imports`` (Boolean, deprecated) equivalent to
-  ``follow_imports=skip`` plus ``ignore_missing_imports=True``.
+Disallow Any
```

Comment:
Maybe give this section a loftier title, e.g. "Disallowing `Any` in various contexts"? Or at least "Disallowing `An`"? I suppose the corresponding section in command_line.rst should also be renamed then, and the label used to link to it.

Reply:
I decided to rename this section and the associated section in the command line page to "Disallowing dynamic typing". (I also renamed the section in the `--help` text to "Dynamic typing").

Hopefully this is more loftier? At the very least, it's a little less jargon-y.